src/allocator_defrag.c

```diff
@@ -0,0 +1,478 @@
+#include <stdio.h>
+#include "zmalloc.h"
+#include "serverassert.h"
+#include "allocator_defrag.h"
+
+#define UNUSED(x) (void)(x)
+
+#if defined(HAVE_DEFRAG) && defined(USE_JEMALLOC)
+
+#define STRINGIFY_(x) #x
+#define STRINGIFY(x) STRINGIFY_(x)
+
+
+#define SLAB_NFREE(out, i) out[(i) * 3]
+#define SLAB_LEN(out, i) out[(i) * 3 + 2]
+#define SLAB_NUM_REGS(out, i) out[(i) * 3 + 1]
+
+#define UTILIZATION_THRESHOLD_FACTOR_MILI (125) // 12.5% additional utilization
+
+/// @brief Helper struct to store MIB (Management Information Base) information for jemalloc bin queries.
+typedef struct je_bin_q_helper {
+    size_t mib_curr_slabs[6];
+    size_t miblen_curr_slabs;
+    size_t mib_nonfull_slabs[6];
+    size_t miblen_nonfull_slabs;
+    size_t mib_curr_regs[6];
+    size_t miblen_curr_regs;
+    size_t mib_nmalloc[6];
+    size_t miblen_nmalloc;
+    size_t mib_ndealloc[6];
+    size_t miblen_ndealloc;
+} je_bin_q_helper;
+
+/// @brief Struct representing bin information.
+typedef struct je_binfo {
```

Comment:
```suggestion
typedef struct je_bin_info {
```
Also must structs in Valkey are camelCase or PascalCase.

Reply:
I'm changing all naming to camelCase